src/main/java/com/mongodb/kafka/connect/sink/dlq/AnalyzedBatchFailedWithBulkWriteException.java

```diff
@@ -108,7 +110,6 @@ public void log() {
     if (!skippedRecords.isEmpty()) {
       logger.log(skippedRecords.values(), writeSkippedException);
     }
-    logger.log(null, e);
```

Comment:
This message was moved to `StartedMongoSinkTask.handleTolerableWriteException` for the following reason: the log message `"Failed to put into the sink the following records: null ... exception details"` is weird and confusing. Specifying a meaningful message from here would require having another method on the `AnalyzedBatchFailedWithBulkWriteException.Logger` interface.

`StartedMongoSinkTask.handleTolerableWriteException` now logs the `MongoBulkWriteException`, without analyzing it and specifying which records actually failed to be written. Then `AnalyzedBatchFailedWithBulkWriteException.log` logs each failed record with a more specific exception relevant to it.